sktime/forecasting/timesfm_forecaster.py

```diff
@@ -343,3 +368,42 @@ def _frame2numpy(data):
         (-1, length, len(data.columns))
     )
     return arr
+
+
+@_multiton
+class _CachedTimesFM:
+    """Cached TimesFM model, to ensure only one instance exists in memory.
+
+    TimesFM is a zero shot model and immutable, hence there will not be
+    any side effects of sharing the same instance across multiple uses.
+    """
+
+    def __init__(self, key, timesfm_kwargs, use_source_package, repo_id):
+        self.key = key
+        self.timesfm_kwargs = timesfm_kwargs
+        self.repo_id = repo_id
+        self.use_source_package = use_source_package
+        self.tfm = None
+
+        self.clients = {}
+
+    def load_from_checkpoint(self, client=None):
+        if client is not None:
+            self.clients += {client}
```

Comment:
I get an error `TypeError: unhashable type: 'TimesFMForecaster'`, this probably should be `+= {client._get_unique_timesfm_key()}`

Reply:
no, this should be the actual instances. I thought this worked with object instances - for me forming sets with instances causes no error, so it might be a python version issue?

Reply:
> no, this should be the actual instances. I thought this worked with object instances - for me forming sets with instances causes no error, so it might be a python version issue?

not working for me either, also its forming a dictionary in line `self.clients = {}` not a set, but its unhashable either way.
also I don't understand why it has be the actual instances?

Reply:
> also I don't understand why it has be the actual instances?

There could be multiple instances with the same parameters. We do not want to destroy the `TimesFM` instance before all the `TimesFMForecasterInstances` have been destroyed.